pkg/lib/regex/regex.go

```diff
@@ -40,3 +40,28 @@ var _alphaNumericDashUnderscoreRegex = regexp.MustCompile(`^[a-zA-Z0-9_\-]+$`)
 func IsAlphaNumericDashUnderscore(s string) bool {
 	return _alphaNumericDashUnderscoreRegex.MatchString(s)
 }
+
+// used the evaluated form of
+// https://github.com/docker/distribution/blob/3150937b9f2b1b5b096b2634d0e7c44d4a0f89fb/reference/regexp.go#L68-L70
+
+var _dockerValidImage = regexp.MustCompile(
+	`^((?:(?:[a-z0-9]|[a-z0-9][a-z0-9-]*[a-z0-9])` +
+		`(?:(?:\.(?:[a-z0-9]|[a-z0-9][a-z0-9-]*[a-z0-9]))+)` +
+		`?(?::[0-9]+)?/)?[a-z0-9]` +
+		`+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)` +
+		`?(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?)` +
+		`(?::([\w][\w.-]{0,127}))` +
+		`?(?:@([A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][[:xdigit:]]{32,}))?$`,
+)
+
+func IsValidDockerImage(s string) bool {
+	return _dockerValidImage.MatchString(s)
+}
+
+var _ecrPattern = regexp.MustCompile(
+	`(^[a-zA-Z0-9][a-zA-Z0-9-_]*)\.dkr\.ecr(\-fips)?\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.amazonaws\.com(\.cn)?`,
+)
+
+func IsValidECR(s string) bool {
+	return _ecrPattern.MatchString(s)
```

Comment:
Should this function also check `IsValidDockerImage()`?

Reply:
Well, it's not actually necessary. When this function is called, we already know that the string we're passing to this function is a valid docker path, because, by that point, it would have already gone through the validation phase. What do you think?

Reply:
So `IsValidECR` goes hand in hand with `IsValidDockerImage`. On the other hand, we could add another 3rd function `IsValidECRImage` to use the previous two. What do you think?